packages/components/format-date/src/format-date.ts

```diff
@@ -0,0 +1,100 @@
+ *
+ * @slot default - A place for the fallback when there is no valid date/time applied.
+ */
+export class FormatDate extends LocaleMixin(LitElement) {
+  #date?: Date = new Date();
+
+  /** The format for displaying the weekday. */
+  @property() weekday?: Intl.DateTimeFormatOptions['weekday'];
+
+  /** The format for displaying the era (eg. 'long' -> `Anno Domini`). */
+  @property() era?: Intl.DateTimeFormatOptions['era'];
+
+  /** The format for displaying the year. Default to `numeric` when all properties are `undefined`. */
+  @property() year?: Intl.DateTimeFormatOptions['year'];
+
+  /** The format for displaying the month. Default to `numeric` when all properties are `undefined`. */
+  @property() month?: Intl.DateTimeFormatOptions['month'];
+
+  /** The format for displaying the day. Default to `numeric` when all properties are `undefined`. */
+  @property() day?: Intl.DateTimeFormatOptions['day'];
+
+  /** The format for displaying the day periods. It only has an effect if a 12-hour clock - hour12 is set to true */
+  @property({ attribute: 'day-period' }) dayPeriod?: Intl.DateTimeFormatOptions['dayPeriod'];
+
+  /** The format for displaying the hour. */
+  @property() hour?: Intl.DateTimeFormatOptions['hour'];
+
+  /** The format for displaying the minute. */
+  @property() minute?: Intl.DateTimeFormatOptions['minute'];
+
+  /** The format for displaying the second. */
+  @property() second?: Intl.DateTimeFormatOptions['second'];
+
+  /** The format for displaying the time. */
+  @property({ attribute: 'time-zone-name' }) timeZoneName?: Intl.DateTimeFormatOptions['timeZoneName'];
+
+  /** The time zone to express the time in. The default is the runtime's default time zone. */
+  @property({ attribute: 'time-zone' }) timeZone?: Intl.DateTimeFormatOptions['timeZone'];
+
+  /** Whether to use 12-hour time or not (when `false` is set 24-hour time is used). The default is locale dependent. */
+  @property({ type: Boolean }) hour12?: Intl.DateTimeFormatOptions['hour12'];
+
+  /**
+   * Use this property if you need access to advanced formatting options not provided via properties of this component.
+   * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat
+   */
+  @property({ type: Object, attribute: 'date-time-options' }) dateTimeOptions?: Intl.DateTimeFormatOptions;
+
+  set date(value: number | string | Date | undefined | null) {
+    const oldValue = this.#date;
+
+    if (value instanceof Date) {
+      this.#date = value;
+    } else if (typeof value === 'number' && this.#isDateValid(value)) {
+      this.#date = new Date(value);
+    } else if (typeof value === 'string' && this.#isDateValid(value)) {
+      this.#date = new Date(value);
+    } else {
+      this.#date = undefined;
+    }
+
+    this.requestUpdate('date', oldValue);
+  }
+
+  /** The date/time to format. If not set, the current date and time will be used. */
+  @property()
```

Comment:
I think that if you move the decorator to the setter, then you don't need the `this.requestUpdate('date', ...)`

Reply:
The docs probably changed at some point. It now shows the decorator on the setter: https://lit.dev/docs/components/properties/#accessors-custom